### DISCUSSION TITLE
Error while deploying the app "reading document at index 1: yaml: unmarshal errors: key "chart" already set in map"

### DISCUSSION THREAD

**Discussion by mohitgith** [helmfile/helmfile#863] [Questions]
When deploying the application in the Kubernetes cluster (AWS EKS v1.26) and with the updated client as `kubectl@1.26.3`, `helm@3.11.3`, and `helmfile@0.153.1`, I'm getting the below issue
```
failed to read install.yaml: reading document at index 1: yaml: unmarshal errors:
  line 195: key "chart" already set in map
  line 202: key "chart" already set in map
  line 209: key "chart" already set in map
  line 216: key "chart" already set in map
  line 223: key "chart" already set in map
  line 230: key "chart" already set in map
  line 237: key "chart" already set in map
  line 244: key "chart" already set in map
  line 251: key "chart" already set in map
  line 258: key "chart" already set in map
  line 265: key "chart" already set in map
  line 272: key "chart" already set in map
  line 279: key "chart" already set in map
  line 286: key "chart" already set in map
  line 293: key "chart" already set in map
  line 307: key "chart" already set in map
  line 326: key "chart" already set in map
  line 333: key "chart" already set in map
  line 340: key "chart" already set in map
  line 347: key "chart" already set in map
  line 354: key "chart" already set in map
```
Here `install.yaml` (configuration) will have a lot of charts to be deployed so, the `chart` have multiple copies but in different anchors.

Everything works fine if we're using `helmfile@1.46.0`

**Comment by yxxhero** [comment]
@mohitgith please show your helmfile. thanks so much.

**Reply by ppenguin** [reply]
I have the same issue, it used to work some months ago, not exactly sure on which versions, but now (with newer versions of `helmfile`, `helm` and `kubectl`, among others) it throws the same error as for OP.

I [posted on SF](https://serverfault.com/questions/1135258/previously-working-helmfile-now-errors-out-with-key-chart-already-set-in-map) but it has gone unanswered there. The related issues I found were interestingly for older versions and marked as solved...

In the link there's also an excerpt of the affected `helmfile`.

**Reply by yxxhero** [reply]
see:https://github.com/helmfile/helmfile/issues/917

**Reply by ppenguin** [reply]
Thanks, setting `HELMFILE_GOCCY_GOYAML=true` did the trick for me
